spring-integration-jms/src/main/java/org/springframework/integration/jms/JmsSendingMessageHandler.java

```diff
@@ -70,6 +70,7 @@ public void setDestinationExpression(Expression destinationExpression) {
 		Assert.isTrue(this.destination == null && this.destinationName == null,
 				"The 'destination', 'destinationName', and 'destinationExpression' properties are mutually exclusive.");
 		this.destinationExpressionProcessor = new ExpressionEvaluatingMessageProcessor<Object>(destinationExpression);
+		setPrimaryExpression(destinationExpression);
```

Comment:
M-m-m. Are you going to add more similar `primaryExpression` manipulations throughout the project?